### ISSUE TITLE
`baseUrl` is blanked on mail send

### ISSUE THREAD

**Issue by julianlam** [sendgrid/sendgrid-nodejs#1353]
### Issue Summary

Apologies for the vague title. I am attempting to debug an installation where @sendgrid/mail fails to send emails. Their configuration is known-working, but seems to have stopped working in recent days.

Upon inspection of the code, I find the following in `@sendgrid/client/src/classes/client.js`

```
options = mergeData(this.defaultRequest, options);
```

Inspecting those two arguments shows that:

* `this.defaultRequest` contains `baseUrl` is `https://api.sendgrid.com`, as expected.
* `options` contains `baseUrl` as `undefined`.

Expectation is that once merged, `baseUrl` will be `https://api.sendgrid.com`.

However, once merged, `baseUrl` becomes `undefined`, **so it looks to be a fault with the merge method in `@sendgrid/helpers`**. However I am not well-versed enough in the code to figure out why.

If I manually set `options.baseUrl = this.defaultRequest.baseUrl;` after the merge, then the mail sends OK.

### Steps to Reproduce
1. This is the first step
2. This is the second step
3. Further steps, etc.

### Code Snippet
```node
# paste code here
```

### Exception/Log
```
uncaughtException: Cannot read properties of null (reading 'length')
TypeError: Cannot read properties of null (reading 'length')
    at isSubdomainOf (/path/to/node_modules/follow-redirects/index.js:563:41)
    at RedirectableRequest._processResponse (/path/to/node_modules/follow-redirects/index.js:396:52)
    at ClientRequest.RedirectableRequest._onNativeResponse (/path/to/node_modules/follow-redirects/index.js:57:10)
    at Object.onceWrapper (node:events:510:26)
    at ClientRequest.emit (node:events:390:28)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:623:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)
    at Socket.socketOnData (node:_http_client:487:22)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
```

### Technical details:
* sendgrid-nodejs version: 7.6.1
* node version: 16.13.0

**Comment by julianlam** [comment]
Hm, this was an odd one, it is resolved once I pin my dependencies to `^7.6.2`. For reference, [these were the old versions](https://github.com/julianlam/nodebb-plugin-emailer-sendgrid/commit/75d890abb34ebba36bdbe14558cd0d13fd1999ff#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519):

```diff
-     "@sendgrid/client": "^7.4.0",
-     "@sendgrid/mail": "^7.0.0",
+     "@sendgrid/client": "^7.6.2",
+     "@sendgrid/mail": "^7.6.2",
```

**Comment by julianlam** [comment]
I think I know the reason why... `@sendgrid/mail` had an old version installed due to lockfile (despite having the `^`), which referenced an old version of `@sendgrid/helpers`, which had a buggy implementation that overwrote the existing value even if the new value was `undefined`. It was fixed at the same time the `request` lib was dropped in favour of `axios`, which is reflected in the `@sendgrid/client` version being v7.4.0.

The mismatch of versions caused this issue.
